extensions/sql-migration/src/api/azure.ts

```diff
@@ -108,6 +202,29 @@ export type SqlVMServer = {
 	tenantId: string,
 	subscriptionId: string
 };
+
+export async function getAvailableSqlDatabaseServers(account: azdata.Account, subscription: Subscription): Promise<AzureSqlDatabaseServer[]> {
+	const api = await getAzureCoreAPI();
+	const path = encodeURI(`/subscriptions/${subscription.id}/providers/Microsoft.Sql/servers?api-version=${SQL_SQLDB_API_VERSION}`);
+	const response = await api.makeAzureRestRequest(account, subscription, path, azurecore.HttpRequestMethod.GET, undefined, true);
+	if (response.errors.length > 0) {
+		throw new Error(response.errors.toString());
+	}
+	sortResourceArrayByName(response.response.data.value);
+	return response.response.data.value;
+}
+
+export async function getAvailableSqlDatabases(account: azdata.Account, subscription: Subscription, resourceGroupName: string, serverName: string): Promise<AzureSqlDatabase[]> {
+	const api = await getAzureCoreAPI();
+	const path = encodeURI(`/subscriptions/${subscription.id}/resourceGroups/${resourceGroupName}/providers/Microsoft.Sql/servers/${serverName}/databases?api-version=${SQL_SQLDB_API_VERSION}`);
+	const response = await api.makeAzureRestRequest(account, subscription, path, azurecore.HttpRequestMethod.GET, undefined, true);
```

Comment:
I would add the 'host' parameter to the `makeAzureRestRequest` call and any other similar calls, like so: https://github.com/microsoft/azuredatastudio/pull/20304/files#diff-4081e8f797712d175859ec2168fe2d2dc901c22daad99608fb7bb77fee2ac3e7R115

Otherwise, it will default to the default host, which only works for the public cloud. This request will fail for any customer on a non-public cloud with this current implementation